.github/workflows/wheels-pure.yml

```diff
@@ -0,0 +1,144 @@
+name: Build RAPIDS project wheels
+
+on:
+  workflow_call:
+    inputs:
+      package-name:
+        required: true
+        type: string
+      package-dir:
+        required: true
+        type: string
+      python-package-cuda-suffix:
+        required: false
+        type: string
+        default: ''
+      python-package-versioneer-override:
+        required: true
+        type: string
+      python-package-build-tag:
+        required: false
+        type: string
+        default: ''
+      test-checkout-src:
+        required: false
+        type: string
+        default: 'true'
+      test-requires:
+        required: false
+        type: string
+        default: ''
+      test-extras:
+        required: false
+        type: string
+        default: ''
+      test-unittest:
+        required: false
+        type: string
+        default: ''
+      test-before:
+        required: false
+        type: string
+        default: 'true'
+
+jobs:
+  wheel-build:
+    name: wheel build pure
+    runs-on: [self-hosted, linux, amd64, cpu4]
+    steps:
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Build pure python wheel with pip
+      uses: rapidsai/shared-action-workflows/purewheel@feat/wheel-ci-actions-2
```

Comment:
Like the comments in purewheel/action.yml, consider absorbing this into this workflow